src/context/ItemsProvider.jsx

```diff
@@ -0,0 +1,48 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { axiosInstance } from "../axios";
+import { UserSessionContext } from "./UserSessionProvider";
+import { CartContext } from "./CartProvider";
+
+export const ItemsContext = createContext(null);
+export function ItemsProvider({ children }) {
+  const [isLoadingContent, setIsLoadingContent] = useState(true);
+  const [items, setItems] = useState([]);
+  const { isLoggedIn } = useContext(UserSessionContext);
+  const { cartItems } = useContext(CartContext);
+
+  const fetchData = async () => {
+    try {
+      const { data } = await axiosInstance.get("/products");
+      setItems(data.data);
+      localStorage.setItem("items", JSON.stringify(data.data));
```

Comment:
You don't need to save the items to localStorage after you set the `items` state, you can use them from there.